### ISSUE TITLE
[BUG] bugs with datetime index

### ISSUE THREAD

**Issue by yaronha** [rapidsai/cudf#835]
**Describe the bug**
When trying to set a dataframe index (of type `timestamp`) using `df.set_index()` there is an error, the same will not fail if the index is set if the dataframe constructor (using index=).

**Steps/Code to reproduce bug**

```python
import cudf
from datetime import datetime
# create a time series
# note: i had to reset the microsec time due to another bug of ns to ms conversion
rng = pd.date_range(end=datetime.now().replace(microsecond=0), periods=4, freq='5s')

# setting the index via init works:
df = cd.DataFrame({'a': range(10,30,5), 'b': range(50,58,2)},index=rng)
print(df)

     a    b
2019-02-01 16:53:45.619048   10   50
2019-02-01 16:53:50.619048   15   52
2019-02-01 16:53:55.619048   20   54
2019-02-01 16:54:00.619048   25   56

# setting index via det_index() fails
# error seem to be caused by the fact timestamp series don't support .all() 
df = cd.DataFrame({'a': range(10,30,5), 'b': range(50,58,2)})
df = df.set_index(rng)

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-44-b4bfb4924612> in <module>
----> 1 df = df.set_index(rng)

/conda/envs/rapids/lib/python3.6/site-packages/cudf-0+unknown-py3.6-linux-x86_64.egg/cudf/dataframe/dataframe.py in set_index(self, index)
    497             df = DataFrame()
    498             for k in self.columns:
--> 499                 df[k] = self[k].set_index(index)
    500             return df
    501 

/conda/envs/rapids/lib/python3.6/site-packages/cudf-0+unknown-py3.6-linux-x86_64.egg/cudf/dataframe/dataframe.py in __setitem__(self, name, col)
    252             self._cols[name] = self._prepare_series_for_add(col)
    253         else:
--> 254             self.add_column(name, col)
    255 
    256     def __delitem__(self, name):

/conda/envs/rapids/lib/python3.6/site-packages/cudf-0+unknown-py3.6-linux-x86_64.egg/cudf/dataframe/dataframe.py in add_column(self, name, data, forceindex)
    619         if isinstance(data, GeneratorType):
    620             data = Series(data)
--> 621         series = self._prepare_series_for_add(data, forceindex=forceindex)
    622         series.name = name
    623         self._cols[name] = series

/conda/envs/rapids/lib/python3.6/site-packages/cudf-0+unknown-py3.6-linux-x86_64.egg/cudf/dataframe/dataframe.py in _prepare_series_for_add(self, col, forceindex)
    595         empty_index = len(self._index) == 0
    596         series = Series(col)
--> 597         if forceindex or empty_index or self._index == series.index:
    598             if empty_index:
    599                 self._index = series.index

/conda/envs/rapids/lib/python3.6/site-packages/cudf-0+unknown-py3.6-linux-x86_64.egg/cudf/dataframe/index.py in __eq__(self, other)
    122         lhs = self.as_column()
    123         rhs = other.as_column()
--> 124         res = lhs.unordered_compare('eq', rhs).all()
    125         return res
    126 

AttributeError: 'Column' object has no attribute 'all'
```
Note that i used set_index() with numeric values and it did work

**Expected behavior**
Setting time based indexes should work like in pandas dataframe

**Environment details (please complete the following information):**
The bug doesnt seem to be environment dependent 

**Additional context**
somewhat related, @tebeka (from our team) reported an issue (#824 ) which also block us from using a simple `df.index = x` 

There is another hidden issue when the range is nanosec based, PyArrow will report ns to ms casting problem (`Casting from timestamp[ns] to timestamp[ms] would lose data`)

**Comment by kkraus14** [comment]
Related to #882

**Comment by kkraus14** [comment]
This was resolved by #720
